lib/vagrant-openshift/helper/command_helper.rb

```diff
@@ -160,7 +160,9 @@ def repo_checkout_bash_command(repo, url)
         git fetch origin
         git fetch --prune origin +refs/tags/*:refs/tags/*
         git reset --merge
-        git checkout #{branch}
+        if ! git checkout #{branch}; then
+          git checkout -b #{branch} origin/#{branch}
```

Comment:
Parts of the git clone logic are over two years old (geard days).  As best I can tell the upstream remote was originally added as part of https://github.com/openshift/vagrant-openshift/pull/64 and then later refactored to the shell based approach we have now in https://github.com/openshift/vagrant-openshift/pull/76.  I'm guessing this command is not only used for jenkins jobs but that some may use it for local development automation and rely on it.  If that were the case they would hit the same problem as me with `git checkout <branch>` if they were using a non-master repo.

If we don't think people are relying on this I would love to rip out the upstream remote logic.  Otherwise, I think we need this patch.  Another approach would be to add yet another option to selectively disable the upstream remote configuration.  I don't really like that because someone would have to hit this same problem before they even knew they needed that flag.